docker-compose/mariadb-config/prod/mariadb.cnf

```diff
@@ -0,0 +1,58 @@
+[mariadb]
+
+port                  = 3306
+bind-address          = 0.0.0.0
+
+# Fine tuning
+max_connections         = 300
+connect_timeout         = 5
+wait_timeout            = 600
+max_allowed_packet      = 32M
+thread_cache_size       = 128
+sort_buffer_size        = 128M
+bulk_insert_buffer_size = 128M
+tmp_table_size          = 32M
+max_heap_table_size     = 32M
+skip-name-resolve       = 1
+innodb_log_file_size    = 2G
+
+# Logging
+# warning log-basename is defined later for replication
+# some log option may be ignored.
+# see: https://mariadb.com/kb/en/mysqld-options/#-log-basename
+# log_error = /var/log/mysql/error.log
+slow_query_log
+slow_query_log_file    = /var/log/mysql/mariadb-slow.log
+long_query_time        = 2
+log_slow_verbosity     = query_plan,explain
+log_error              = /var/log/mysql/mariadb.err.log
+
+# Query cache
+query_cache_size        = 16M
+
+# Character sets
+character-set-server = utf8mb4
+collation-server     = utf8mb4_general_ci
+
+# InnoDB
+# InnoDB is enabled by default with a 10MB datafile in /var/lib/mysql/.
+# Read the manual for more InnoDB related options. There are many!
+innodb_buffer_pool_size = 40G
```

Comment:
Good. Can check with `show global status like 'innodb_buffer_pool_read%'` on the buffer pool performance.

All options that aren't innodb, log_slow_query, or replication based are either default or potentially harmful (like query_cache_size).

Adjust as needed based on observations, but original default bloated config file was a bit harmfull and the updates to get to 10.11 made the default server values much better. When you need to change a setting, do so and comment why in commit/comment.

Reply:
For Production, where we currently have 40G.
The `Innodb_buffer_pool_read_requests/Innodb_buffer_pool_reads = 1043` ratio does tell that we are in a good state, right?


```
MariaDB [(none)]> show global status like 'innodb_buffer_pool_read%';
+---------------------------------------+-------------+
| Variable_name                         | Value       |
+---------------------------------------+-------------+
| Innodb_buffer_pool_read_ahead_rnd     | 0           |
| Innodb_buffer_pool_read_ahead         | 3866961     |
| Innodb_buffer_pool_read_ahead_evicted | 0           |
| Innodb_buffer_pool_read_requests      | 92104167464 |
| Innodb_buffer_pool_reads              | 88307035    |
+---------------------------------------+-------------+
```

Reply:
yep. Very good.